File: exemplos do livro 'Desenvolvendo com C#'/Apêndice F/Linguagens/MinhaClasse.cpp
//cl /LD /CLR /FeMinhaClasseCPP.dll MinhaClasse.cpp

#using <mscorlib.dll>
#using <MinhaClasseCS.dll>
#using <MinhaClasseVB.dll>

__gc public class MinhaClasseCPP : public MinhaClasseVB
{

public: 
  void Metodo3()
  {
    System::Console::WriteLine("VC++.NET");
  }
};

File: exemplos do livro 'Desenvolvendo com C#'/Apêndice F/Linguagens/MinhaClasse.vb
'vbc /t:library /out:MinhaClasseVB.dll MinhaClasse.vb /r:MinhaClasse.dll

Public Class MinhaClasseVB 
  Inherits MinhaClasseCS

  Public Sub Metodo2() 
  
    System.Console.WriteLine("VB.NET")
  
  End Sub

End Class